# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ModuleLoadRequest.cpp
# Commit: d52858ae58d2
# Full Hash: d52858ae58d25708320093094a8ff791eacf2b15
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-21 21:39:46
# Regressor Bug: 1742437
# File Overlap Count: 1
# Description:
#   Bug 1742437 - Split ScriptLoadRequest into ScriptLoadRequest and DOMScriptLoadContext; r=jonco,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D134041
# ==============================================================================

diff -r 0355b14a5461 -r d52858ae58d2 dom/script/ModuleLoadRequest.cpp
--- a/dom/script/ModuleLoadRequest.cpp	Mon Feb 21 14:53:59 2022 +0000
+++ b/dom/script/ModuleLoadRequest.cpp	Mon Feb 21 15:39:04 2022 +0000
@@ -54,10 +54,13 @@
 ModuleLoadRequest* ModuleLoadRequest::CreateTopLevel(
     nsIURI* aURI, ScriptFetchOptions* aFetchOptions,
     const SRIMetadata& aIntegrity, nsIURI* aReferrer, ScriptLoader* aLoader) {
-  return new ModuleLoadRequest(
+  auto* request = new ModuleLoadRequest(
       aURI, aFetchOptions, aIntegrity, aReferrer, true, /* is top level */
       false,                                            /* is dynamic import */
       aLoader->GetModuleLoader(), NewVisitedSetForTopLevelImport(aURI));
+  DOMScriptLoadContext* context = new DOMScriptLoadContext(request);
+  request->mLoadContext = context;
+  return request;
 }
 
 /* static */
@@ -68,9 +71,11 @@
                             aParent->mURI, false, /* is top level */
                             false,                /* is dynamic import */
                             aParent->mLoader, aParent->mVisitedSet);
+  DOMScriptLoadContext* context = new DOMScriptLoadContext(request);
+  context->mIsInline = false;
+  context->mScriptMode = aParent->GetLoadContext()->mScriptMode;
 
-  request->mIsInline = false;
-  request->mScriptMode = aParent->mScriptMode;
+  request->mLoadContext = context;
 
   return request;
 }
@@ -88,8 +93,10 @@
       true, /* is dynamic import */
       aLoader->GetModuleLoader(), NewVisitedSetForTopLevelImport(aURI));
 
-  request->mIsInline = false;
-  request->mScriptMode = ScriptMode::eAsync;
+  DOMScriptLoadContext* context = new DOMScriptLoadContext(request);
+  context->mIsInline = false;
+  context->mScriptMode = DOMScriptLoadContext::ScriptMode::eAsync;
+  request->mLoadContext = context;
   request->mDynamicReferencingPrivate = aReferencingPrivate;
   request->mDynamicSpecifier = aSpecifier;
   request->mDynamicPromise = aPromise;